subsys/bluetooth/shell/gatt.c

```diff
@@ -478,19 +478,88 @@ static int cmd_unsubscribe(const struct shell *shell,
 }
 #endif /* CONFIG_BT_GATT_CLIENT */
 
+static struct db_stats {
+	u16_t svc_count;
+	u16_t attr_count;
+	u16_t chrc_count;
+	u16_t ccc_count;
+	size_t ccc_cfg;
+} stats;
+
+struct show_data {
+	const struct shell *shell;
+	struct bt_uuid_16 uuid;
+};
+
 static u8_t print_attr(const struct bt_gatt_attr *attr, void *user_data)
 {
-	const struct shell *shell = user_data;
+	struct show_data *data = user_data;
+
+	if (data->uuid.val) {
+		if (!bt_uuid_cmp(&data->uuid.uuid, attr->uuid)) {
+			goto print;
+		}
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	stats.attr_count++;
+
+	if (!bt_uuid_cmp(attr->uuid, BT_UUID_GATT_PRIMARY) ||
+	    !bt_uuid_cmp(attr->uuid, BT_UUID_GATT_SECONDARY)) {
+		stats.svc_count++;
+	}
+
+	if (!bt_uuid_cmp(attr->uuid, BT_UUID_GATT_CHRC)) {
+		stats.chrc_count++;
+	}
 
-	shell_print(shell, "attr %p handle 0x%04x uuid %s perm 0x%02x",
-	      attr, attr->handle, bt_uuid_str(attr->uuid), attr->perm);
+	if (!bt_uuid_cmp(attr->uuid, BT_UUID_GATT_CCC) &&
+	    attr->write == bt_gatt_attr_write_ccc) {
+		struct _bt_gatt_ccc *cfg = attr->user_data;
+
+		stats.ccc_count++;
+		stats.ccc_cfg += cfg->cfg_len;
+	}
+
+print:
+	shell_print(data->shell, "attr %p handle 0x%04x uuid %s perm 0x%02x",
+		    attr, attr->handle, bt_uuid_str(attr->uuid), attr->perm);
 
 	return BT_GATT_ITER_CONTINUE;
 }
 
 static int cmd_show_db(const struct shell *shell, size_t argc, char *argv[])
 {
-	bt_gatt_foreach_attr(0x0001, 0xffff, print_attr, (void *)shell);
+	struct show_data data;
+	size_t total_len;
+
+	memset(&stats, 0, sizeof(stats));
+
+	data.shell = shell;
+
+	if (argc > 1) {
+		data.uuid.uuid.type = BT_UUID_TYPE_16;
+		data.uuid.val = strtoul(argv[1], NULL, 16);
+	} else {
+		data.uuid.val = 0;
+	}
+
+	bt_gatt_foreach_attr(0x0001, 0xffff, print_attr, &data);
+
+	if (!stats.attr_count) {
+		return 0;
```

Comment:
Maybe some feedback cound be printed here?

Reply:
I will add a message when no attribute are found.